problem_builder/tests/integration/test_mentoring.py

```diff
@@ -226,32 +290,40 @@ def test_feedbacks_and_messages_is_not_shown_on_first_load(self):
         self.assertFalse(messages.is_displayed())
         self.assertFalse(submit.is_enabled())
 
-    def test_persists_feedback_on_page_reload(self):
-        mentoring = self.load_scenario("feedback_persistence.xml")
-        answer, mcq, mrq, rating = self._get_controls(mentoring)
-        messages = self._get_messages_element(mentoring)
-
-        self._standard_filling(answer, mcq, mrq, rating)
-        self.click_submit(mentoring)
-        self._standard_checks(answer, mcq, mrq, rating, messages)
-
-        # now, reload the page and do the same checks again
-        mentoring = self.reload_student_view()
-        answer, mcq, mrq, rating = self._get_controls(mentoring)
-        messages = self._get_messages_element(mentoring)
-        submit = mentoring.find_element_by_css_selector('.submit input.input-main')
-
-        self._standard_checks(answer, mcq, mrq, rating, messages)
-        # after reloading submit is disabled...
-        self.assertFalse(submit.is_enabled())
-
-        # ...until some changes are done
-        self.click_choice(mrq, "Its elegance")
-        self.assertTrue(submit.is_enabled())
-
-    def test_does_not_persist_mcq_feedback_on_page_reload_if_disabled(self):
+    @ddt.data(
+        (
+            {
+                'pb_mcq_hide_previous_answer': False,
+                'pb_hide_feedback_if_attempts_remain': False
+            },
+            "_standard_checks"
+        ),
+        (
+            {
+                'pb_mcq_hide_previous_answer': False,
+                'pb_hide_feedback_if_attempts_remain': True
+            },
+            "_hide_feedback_checks"
+        ),
+        (
+            {
+                'pb_mcq_hide_previous_answer': True,
+                'pb_hide_feedback_if_attempts_remain': False
+            },
+            "_mcq_hide_previous_answer_checks"
+        ),
+        (
+            {
+                'pb_mcq_hide_previous_answer': True,
+                'pb_hide_feedback_if_attempts_remain': True
+            },
+            "_mcq_hide_previous_answer_hide_feedback_checks"
+        ),
+    )
+    @ddt.unpack
+    def test_persistence(self, options, after_reload_checks):
```

Comment:
DDT stands for data driven testing, I think that names of methods that perform assertions don't qualify as "data" :). I think that splitting this into a helper method (very similar to ``test_persistence``) and four methods that launch it would be beneficial in this case to the overall clarity.